docs/PATH_NOTATION.md

```diff
@@ -65,14 +65,209 @@ set({}, 'nested.1', 'new value')
 // }
 ```
 
-## :construction: This page is still a work in progress, the following sections will be written soon
-
 ## Bracket notation (property access)
+
+The bracket notation is the same as JavaScript [bracket notation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_Accessors). It gives access to properties with an invalid identifier or dots in it.
+
+```js
+set({}, 'prop.[nested.1]', 'new value')
+// Returns:
+// {
+//   "prop": {
+//     "nested.1": "new value"
+//   }
+// }
+```
+
 ## Array access notation
+
+Similar to the JavaScript way to access elements of an array, gives an easy access to deeply nested elements:
+
+```js
+set({}, 'prop.nested[0]', 'new value')
+// Returns:
+// {
+//   "prop": {
+//     "nested": [
+//       "new value"
+//     ]
+//   }
+// }
+``` 
+
 ## Property list notation
+
+We will use an initial object for the following examples: 
+
+```js
+// {
+//   "prop": {
+//     "nested": {
+//       "0": 'old value',
+//       "1": 'old value',
+//       "2": 'old value'
+//     }
+//   }
+// }
+```
+
+
+The list notation provides a way to apply operation on multiple properties of an objet:
+
+```js
+set(initial, 'prop.nested{0, 1}', 'new value')
```

Comment:
If I'm not mistaken spaces after commas aren't supported, so this would update properties "0" and " 1".
Why don't you use property names with letters such as "foo" "bar", this would avoid mistaking with array indexes.